integration_test/test_videoroom/test/integration/simulcast_test.exs

```diff
@@ -0,0 +1,372 @@
+        | id: browser_id,
+          buttons: buttons,
+          start_buttons: start_buttons
+      }
+
+      Task.async(fn ->
+        Stampede.start({SimulcastMustang, specific_mustang}, @browser_options)
+      end)
+    end
+    |> Task.await_many(:infinity)
+
+    receive do
+      acc ->
+        for stage <- stages,
+            browsers = Map.get(acc, stage) do
+          encoding =
+            case stage do
+              :disable -> "h"
+              :enable -> "m"
+            end
+
+          sender = browsers[0]
+          receiver = browsers[1]
+          stats = Enum.zip(sender, receiver)
+
+          assert(
+            Enum.any?(stats, fn {sender, receiver} ->
+              stats_for_encoding?(receiver, sender, encoding)
+            end),
+            "Failed on stage: #{stage} should be encoding: #{encoding},
+                receiver stats are: #{inspect(receiver)}
+                sender stats are: #{inspect(Enum.map(sender, & &1[encoding]))}
+                "
+          )
+        end
+    end
+  end
+
+  @moduletag timeout: @test_duration
+  test "User change encoding to low and then return to medium" do
+    browsers_number = 2
+
+    pid = self()
+
+    receiver = Process.spawn(fn -> receive_stats(browsers_number, pid) end, [:link])
+
+    mustang_options = %{
+      target_url: @room_url,
+      linger: 40_000,
+      join_interval: @join_interval,
+      start_button: @start_with_simulcast,
+      receiver: receiver,
+      buttons: [],
+      receiver_button: @simulcast_inbound_stats,
+      sender_button: @simulcast_outbound_stats,
+      start_buttons: [],
+      id: -1
+    }
+
+    create_room(mustang_options)
+
+    timeouts = List.duplicate(@step_duration, 2)
+
+    buttons1 =
+      [
+        @set_peer_encoding_low,
+        @set_peer_encoding_medium
+      ]
+      |> Enum.zip(timeouts)
+
+    buttons2 =
+      [
+        @simulcast_outbound_stats,
+        @simulcast_outbound_stats
+      ]
+      |> Enum.zip(timeouts)
+
+    stages = [:disable, :enable]
+
+    buttons_with_id =
+      [buttons1, buttons2]
+      |> Enum.map(&Enum.zip(&1, stages))
+      |> Enum.with_index()
+      |> Map.new(fn {button, browser_id} -> {browser_id, button} end)
+
+    for {browser_id, buttons} <- buttons_with_id, into: [] do
+      start_buttons = if browser_id == 0, do: @start_buttons, else: []
+
+      specific_mustang = %{
+        mustang_options
+        | id: browser_id,
+          buttons: buttons,
+          start_buttons: start_buttons
+      }
+
+      Task.async(fn ->
+        Stampede.start({SimulcastMustang, specific_mustang}, @browser_options)
+      end)
+    end
+    |> Task.await_many(:infinity)
+
+    receive do
+      acc ->
+        for stage <- stages,
+            browsers = Map.get(acc, stage) do
+          encoding =
+            case stage do
+              :disable -> "l"
+              :enable -> "m"
+            end
+
+          receiver = browsers[0]
+          sender = browsers[1]
```

Comment:
Could we name `acc`, `browsers` `receiver` and `sender` in a more meaningful way? It's pretty hard to understand what they store especially that we pass `sender` to `stats_for_encoding?` as `sender_stats`. Inside `stats_for_encoding` we convert `sender_stats` to `sender`. This makes everything difficult to understand.